default/methods/source_control/file/file_browser.py

```diff
@@ -575,55 +612,58 @@ def __build_media_type_value(self) -> str:
             media_type_query = 'audio'
         return media_type_query
 
-    def file_view_core(
-        self,
-        mode = "serialize"):
-        """
-        mode
-            serialize is in context of web, ie serialize the resulting list
-                currently defaults to this context
-            objects returns the database objects, ie for auto commit
-
-        """
-        output_file_list = []
-        limit_counter = 0
-        file_count = 0  # File count includes ones we don't actually query
-        # outside of limits...
-
-        if self.metadata['file_view_mode'] is None or \
-            self.metadata['file_view_mode'] not in ["changes", "annotation", "home", "task", "explorer", "base", 'ids_only']:
-            self.log['error']['file_view_mode'] = 'Invalid file_view_mode "{}"'.format(self.metadata['file_view_mode'])
-            return None
-        ignore_id_list = None
+    def get_query_file_list(self):
```

Comment:
I've pulled apart the `file_view_core` function into smaller functions, while trying to keep all logic the same. `file_view_core` still exists and does the same thing, but it calls a set of other functions instead of keeping all code inline.

I tried to leave all logic the same, and tested as much as I could, but someone who understands the full system should do a good QA to ensure I didnt break anything.

Reply:
The `get_query_file_list` method replaces [this block](https://github.com/diffgram/diffgram/blob/38d12c685c47a53061bcbe9b80c1ae257e651a84/default/methods/source_control/file/file_browser.py#L668-L675) of code